webapp/src/main/java/com/box/l10n/mojito/service/scheduledjob/jobs/ScheduledThirdPartySync.java

```diff
@@ -0,0 +1,146 @@
+import org.springframework.web.util.UriComponentsBuilder;
+
+/**
+ * Scheduled job that performs the third party sync for a repository off a cron schedule.
+ *
+ * @author mattwilshire
+ */
+@Component
+@DisallowConcurrentExecution
+public class ScheduledThirdPartySync implements IScheduledJob {
+
+  static Logger logger = LoggerFactory.getLogger(ScheduledThirdPartySync.class);
+
+  @Autowired private QuartzPollableTaskScheduler quartzPollableTaskScheduler;
+  @Autowired private ScheduledJobRepository scheduledJobRepository;
+  @Autowired private PagerDutyIntegrationService pagerDutyIntegrationService;
+
+  @Autowired ServerConfig serverConfig;
+
+  @Value(
+      "${l10n.scheduledJobs.thirdPartySync.notifications.title:MOJITO | Third party sync failed for {repository}}")
+  String notificationTitle;
+
+  private ScheduledJob scheduledJob;
+  private Long pollableTaskId;
+
+  @Override
+  public void execute(JobExecutionContext jobExecutionContext) throws JobExecutionException {
+    // Fetch the scheduled job and cast the properties
+    scheduledJob = scheduledJobRepository.findByJobKey(jobExecutionContext.getJobDetail().getKey());
+    ScheduledThirdPartySyncProperties scheduledJobProperties =
+        (ScheduledThirdPartySyncProperties) scheduledJob.getProperties();
+
+    logger.info(
+        "Third party sync for repository {} has started.", scheduledJob.getRepository().getName());
+
+    // Create ThirdPartySyncInput from scheduled job and properties
+    ThirdPartySyncJobInput thirdPartySyncJobInput =
+        new ThirdPartySyncJobInput(scheduledJob, scheduledJobProperties);
+
+    try {
+      PollableFuture<Void> task =
+          quartzPollableTaskScheduler.scheduleJobWithCustomTimeout(
+              ThirdPartySyncJob.class, thirdPartySyncJobInput, "thirdPartySync", 3600L);
+      pollableTaskId = task.getPollableTask().getId();
+      // Wait for sync to complete
+      task.get();
+    } catch (Exception e) {
+      throw new JobExecutionException(e);
+    }
+  }
+
+  @Override
+  public void onSuccess(JobExecutionContext context) {
+    logger.info(
+        "Third-Party Sync succeeded for repository {}.", scheduledJob.getRepository().getName());
+
+    // Resolve PD incident
+    pagerDutyIntegrationService
+        .getDefaultPagerDutyClient()
+        .ifPresent(
+            pd -> {
+              try {
+                pd.resolveIncident(scheduledJob.getId());
+              } catch (PagerDutyException e) {
+                logger.error(
+                    "Couldn't send resolve PagerDuty notification for successful third party sync of repository: '{}' -",
+                    scheduledJob.getRepository().getName(),
+                    e);
+              }
+            });
+  }
+
+  @Override
+  public void onFailure(JobExecutionContext context, JobExecutionException jobException) {
+    logger.error(
+        "Third-Party Sync for repository {} has failed. Pollable Task ID: {}",
```

Comment:
I just want to confirm that this pollable task id is the id stored in the db? It's not some transient id used only while the job is running?

Reply:
Yes, all pollable task ids get stored in the db table 'pollable_task'.